crates/config/src/zksync.rs

```diff
@@ -125,7 +126,8 @@ impl ZkSyncConfig {
         let optimizer = Optimizer {
             enabled: Some(self.optimizer),
             mode: Some(self.optimizer_mode),
-            fallback_to_optimizing_for_size: Some(self.fallback_oz),
+            size_fallback: Some(self.should_compile()),
```

Comment:
does this mean it's _always_ enabled?

Reply:
No, good catch. My fault, this should say self.size_fallback